xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/LegacyLedgerSpecifierUtils.java

```diff
@@ -0,0 +1,45 @@
+package org.xrpl.xrpl4j.model.client;
+
+import org.xrpl.xrpl4j.model.client.common.LedgerIndex;
+import org.xrpl.xrpl4j.model.client.common.LedgerIndexShortcut;
+import org.xrpl.xrpl4j.model.client.common.LedgerSpecifier;
+import org.xrpl.xrpl4j.model.transactions.Hash256;
+
+import java.util.Optional;
+
+/**
+ * Simple utility class to help deprecate ledger index and ledger hash fields in request parameter objects.
+ *
+ * @deprecated This will be removed once all {@code ledgerIndex} and {@code ledgerHash} fields are removed.
+ */
+@Deprecated
+public class LegacyLedgerSpecifierUtils {
+
+  /**
+   * Computes a {@link LedgerSpecifier} from a ledger hash and a ledger index.
+   *
+   * @param ledgerHash An optionally-present {@link Hash256}.
+   * @param ledgerIndex A {@link LedgerIndex}.
+   * @return A {@link LedgerSpecifier}.
+   */
+  public static LedgerSpecifier computeLedgerSpecifier(Optional<Hash256> ledgerHash, LedgerIndex ledgerIndex) {
+    return ledgerHash
+      .map(LedgerSpecifier::ledgerHash)
+      .orElseGet(() -> {
+        if (ledgerIndex != null) {
+          if (ledgerIndex.equals(LedgerIndex.VALIDATED)) {
+            return LedgerSpecifier.VALIDATED;
+          } else if (ledgerIndex.equals(LedgerIndex.CURRENT)) {
+            return LedgerSpecifier.CURRENT;
```

Comment:
We should increase test coverage here.

Reply:
see https://github.com/XRPLF/xrpl4j/pull/126/files#diff-eb3ba4fa4fb59d2592f632aeffdd45be98ed34e9cb79c3e658831745e29ca24b